### ISSUE TITLE
Feature Request: don't report on unused parameters for deprecated functions

### ISSUE THREAD

**Issue by jrfnl** [sirbrillig/phpcs-variable-analysis#302]
```php
/**
 * Function description.
 *
 * @deprecated #.#
 */
function foo($old_param, $another_old_param) {
    trigger_error('Deprecated function, use ... instead', E_USER_DEPRECATED);
    something_else($old_param);
}
```

The above code sample would trigger the following warning:
```
 8 | WARNING | Unused function parameter $another_old_param.
   |         | (VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable)
```

... while I would expect no warning as the function is deprecated.
